### PR TITLE
Provide a convenient way to install (and use Kubernetes secrets) #18

### PR DESCRIPTION
**Pull request by collimarco**
Pull request for issue #18

### CODE REVIEW

**Comment by yyyogev** [comment]
@imnotashrimp please review the change in the README

**Comment by idohalevi** [comment]
@collimarco 
I'll test it later this week or at the beginning of next week and release a new version

**Comment by idohalevi** [comment]
@collimarco looks good to me thank you for the PR.
@imnotashrimp  can you please review the readme?

File: README.md [hunk 1]
```diff
@@ -19,6 +19,18 @@ The DaemonSet yaml file has two environment variables. Fluentd uses these variab
 |   LOGZIO_TOKEN       | [Logz.io account token](https://app.logz.io/#/dashboard/settings/general) |
 |   LOGZIO_URL         | Logz.io listener url, If the account is in the EU region insert https://listener-eu.logz.io:8071. Otherwise, use https://listener.logz.io:8071. You can tell your account's region by checking your login URL - app.logz.io means you are in the US. app-eu.logz.io means you are in the EU |
 
+For improved security you should store the variables using Kubernetes secrets:
+
+```
+kubectl create secret generic logzio-secrets --from-literal=logzio_token='MY_LOGZIO_TOKEN' --from-literal=logzio_url='MY_LOGZIO_URL' -n kube-system
```
Comment (imnotashrimp) [inline README.md]: @idohalevi I have a draft for k8s metrics that's not quite out the door. Thinking ahead here—should we change variable names here to make sure secrets for logs and metrics are discretely named?

That way, users can have both log and metrics daemonsets installed—with their secrets—in k8s without issue. That should also keep users from running into naming conflicts if they don't modify our default config files.

File: logzio-daemonset-rbc.yaml [hunk 2]
```diff
@@ -62,10 +62,16 @@ spec:
       - name: fluentd
         image: logzio/logzio-k8s:latest
         env:
-        - name:  LOGZIO_TOKEN
-          value: "your logz.io account token"
-        - name:  LOGZIO_URL
-          value: "your logz.io host url"
+        - name: LOGZIO_TOKEN
```
Comment (imnotashrimp) [inline logzio-daemonset-rbc.yaml]: @idohalevi Same note here re: different environment variable names for logs and metrics.

File: README.md (hunk 1 above)
Comment (imnotashrimp) [reply README.md]: @idohalevi Also, same note on calling the secret `logzio-secrets`. Maybe renaming to `logzio-logs-secret` or something like that.

**Comment by imnotashrimp** [comment]
@collimarco Thanks for the PR! By no fault of your own, this may conflict with something else I'm working on. I'll take @idohalevi's lead on whether to merge this now and fix the conflicts, or whether we can fix this all quickly now before merging.

File: README.md (hunk 1 above)
Comment (imnotashrimp) [inline README.md]:
```suggestion
kubectl create secret generic logzio-logs-secret --from-literal=logzio-log-shipping-token='<<ACCOUNT-TOKEN>>' --from-literal=logzio-log-listener-host='<<LISTENER-HOST>>' -n kube-system
```

File: logzio-daemonset-rbc.yaml (hunk 2 above)
Comment (imnotashrimp) [inline logzio-daemonset-rbc.yaml]:
```suggestion
        - name: LOGZIO_LOG_SHIPPING_TOKEN
```

File: logzio-daemonset-rbc.yaml [hunk 3]
```diff
@@ -62,10 +62,16 @@ spec:
...
+          valueFrom:
+            secretKeyRef:
+              name: logzio-secrets
+              key: logzio_token
+        - name: LOGZIO_URL
```
Comment (imnotashrimp) [inline logzio-daemonset-rbc.yaml]:
```suggestion
        - name: LOGZIO_LOG_LISTENER_HOST
```

File: logzio-daemonset-rbc.yaml [hunk 4]
```diff
@@ -62,10 +62,16 @@ spec:
...
+          valueFrom:
+            secretKeyRef:
+              name: logzio-secrets
```
Comment (imnotashrimp) [inline logzio-daemonset-rbc.yaml]:
```suggestion
              name: logzio-logs-secret
```

File: logzio-daemonset-rbc.yaml [hunk 5]
```diff
@@ -62,10 +62,16 @@ spec:
...
         image: logzio/logzio-k8s:latest
         env:
-        - name:  LOGZIO_TOKEN
-          value: "your logz.io account token"
-        - name:  LOGZIO_URL
-          value: "your logz.io host url"
+        - name: LOGZIO_TOKEN
+          valueFrom:
+            secretKeyRef:
+              name: logzio-secrets
```
Comment (imnotashrimp) [inline logzio-daemonset-rbc.yaml]:
```suggestion
              name: logzio-logs-secret
```

File: logzio-daemonset-rbc.yaml [hunk 6]
```diff
@@ -62,10 +62,16 @@ spec:
...
+              key: logzio_token
```
Comment (imnotashrimp) [inline logzio-daemonset-rbc.yaml]:
```suggestion
              key: logzio-log-shipping-token
```

File: logzio-daemonset-rbc.yaml [hunk 7]
```diff
@@ -62,10 +62,16 @@ spec:
...
+        - name: LOGZIO_URL
+          valueFrom:
+            secretKeyRef:
+              name: logzio-secrets
+              key: logzio_url
```
Comment (imnotashrimp) [inline logzio-daemonset-rbc.yaml]:
```suggestion
              key: logzio-log-listener-host
```

File: logzio-daemonset.yaml [hunk 8]
```diff
@@ -22,10 +22,16 @@ spec:
       - name: fluentd
         image: logzio/logzio-k8s:latest
         env:
-        - name:  LOGZIO_TOKEN
-          value: "your logz.io account token"
-        - name:  LOGZIO_URL
-          value: "your logz.io host url"
+        - name: LOGZIO_TOKEN
```
Comment (imnotashrimp) [inline logzio-daemonset.yaml]:
```suggestion
        - name: LOGZIO_LOG_SHIPPING_TOKEN
```

File: logzio-daemonset.yaml [hunk 9]
```diff
@@ -22,10 +22,16 @@ spec:
...
+          valueFrom:
+            secretKeyRef:
+              name: logzio-secrets
```
Comment (imnotashrimp) [inline logzio-daemonset.yaml]:
```suggestion
              name: logzio-logs-secret
```

File: logzio-daemonset.yaml [hunk 10]
```diff
@@ -22,10 +22,16 @@ spec:
...
+              key: logzio_token
+        - name: LOGZIO_URL
+          valueFrom:
+            secretKeyRef:
+              name: logzio-secrets
```
Comment (imnotashrimp) [inline logzio-daemonset.yaml]:
```suggestion
              name: logzio-logs-secret
```

File: logzio-daemonset.yaml [hunk 11]
```diff
@@ -22,10 +22,16 @@ spec:
...
         env:
-        - name:  LOGZIO_TOKEN
-          value: "your logz.io account token"
-        - name:  LOGZIO_URL
-          value: "your logz.io host url"
+        - name: LOGZIO_TOKEN
+          valueFrom:
+            secretKeyRef:
+              name: logzio-secrets
+              key: logzio_token
```
Comment (imnotashrimp) [inline logzio-daemonset.yaml]:
```suggestion
              key: logzio-log-shipping-token
```

File: logzio-daemonset.yaml [hunk 12]
```diff
@@ -22,10 +22,16 @@ spec:
...
+        - name: LOGZIO_URL
+          valueFrom:
+            secretKeyRef:
+              name: logzio-secrets
+              key: logzio_url
```
Comment (imnotashrimp) [inline logzio-daemonset.yaml]:
```suggestion
              key: logzio-log-listener-host
```

File: logzio-daemonset.yaml [hunk 13]
```diff
@@ -22,10 +22,16 @@ spec:
...
-        - name:  LOGZIO_TOKEN
-          value: "your logz.io account token"
-        - name:  LOGZIO_URL
-          value: "your logz.io host url"
+        - name: LOGZIO_TOKEN
+          valueFrom:
+            secretKeyRef:
+              name: logzio-secrets
+              key: logzio_token
+        - name: LOGZIO_URL
```
Comment (imnotashrimp) [inline logzio-daemonset.yaml]:
```suggestion
        - name: LOGZIO_LOG_LISTENER_HOST
```

**Comment by collimarco** [comment]
@imnotashrimp Thanks for taking time on this... However why have you changed all the names in longer and more obscure names? For example `LOGZIO_TOKEN` and `LOGZIO_URL` were pretty clear and already used extensively in your code and docs.

**Comment by imnotashrimp** [comment]
@collimarco This is part of a broader effort to align our docs and make them more consistent with each other. 

In the past, there was some confusion why we used `URL` when we mean just the host. Same with tokens— we now have different kinds of tokens our system uses (API, shared token, account/shipping token). So you can imagine it would get confusing when we ask for a token without making it clear which one a user needs to provide.

Hope this clarifies.

**Comment by idohalevi** [comment]
@collimarco Can you commit the requested changes?

**Comment by yyyogev** [comment]
Migrated to a [local PR](https://github.com/logzio/logzio-k8s/pull/23)

**Pull request closed without merging**